File: 🟢 Easy/1331 - Rank Transform of an Array/_1331RankTransformofanArray.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        set<int> arrSet(arr.begin(), arr.end());

        unordered_map<int, int> map;
        int r = 1;
        for (auto& el : arrSet) {
            map.insert({el, r});
            r++;
        }

        for (int i = 0; i < arr.size(); i++) {
            arr[i] = map[arr[i]];
        }

        return arr;
    }
};

File: 🟢 Easy/1331 - Rank Transform of an Array/_1331RankTransformofanArray.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        setArr = sorted(set(arr))

        mapForRank = {}
        r = 1

        for el in setArr:
            mapForRank[el] = r
            r += 1

        for i in range(0, len(arr)):
            arr[i] = mapForRank[arr[i]]

        return arr
